### ISSUE TITLE
Node.js 8 + npm 5 install Gulp 4: No matching version found for undefined@4.0

### ISSUE THREAD

**Issue by albrin** [npm/npm#16864]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
Gulp version 4.0 is not installed after the update of npm (to version 5).
#### How can the CLI team reproduce the problem?
````
npm install --save-dev github:gulpjs/gulp#4.0
````
I get an error:
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for undefined@4.0
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Alexander\AppData\Roaming\npm-cache\_logs\2017-06-01T11_27_23_145Z-debug.log
```
https://gist.github.com/albrin/1f9778b9abb5e5a1b231e4946802250b

### supporting information:

 - `npm -v` prints: 5.0.1
 - `node -v` prints: v8.0.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by simonua** [comment]
Similar problem here but with different error output:

STRs:
`md npm501 && cd npm501 && npm init -y && npm i gulp@gulpjs/gulp#4.0`

npm **5.0.1**
node **v8.0.0**

Error Message:
**npm ERR! exited with error code: 128**

https://gist.github.com/simonua/e3648b1dad158fc0423ec339cf0d1da2

**Comment by simonua** [comment]
The STRs succeed when using npm@5.0.0.

**Comment by RomainLanz** [comment]
Same here.

**Comment by hmalphettes** [comment]
It is a regression introduced when pacote was upgraded from 2.7.24 to 2.7.26

pacote already has a new release - 2.7.27 - with one change titled `fix(git): fix semver range detection. oops`: https://github.com/zkat/pacote/commit/76d92331eaf2371cf6205eb031fe59b4d70e86fe

I replaced /usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/git.js
by the fixed file:
https://raw.githubusercontent.com/zkat/pacote/76d92331eaf2371cf6205eb031fe59b4d70e86fe/lib/fetchers/git.js
and it fixes the issue at hand.

cc @zkat  

I am excited at how much faster npm-5 feels and so far this is the only bump in the road for me.
I hope this helps!

**Comment by simonua** [comment]
Nice forensics and work-around, @hmalphettes! I confirmed this works for me, too, after editing the file on Windows 7:

`%userprofile%\AppData\Roaming\npm\node_modules\npm\node_modules\pacote\lib\fetchers\git.js`

Looking forward to this being resolved for good in npm@5.0.2. =P

**Comment by zkat** [comment]
This is fixed in [`npmc`](https://npm.im/npmc) and will be available ones `npm@5.0.2` is released.
